bundle-workflow/src/ci_workflow/check_gradle_dependencies.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import re
+
+from ci_workflow.check import Check
+from system.properties_file import PropertiesFile
+
+
+class CheckGradleDependencies(Check):
+    def __init__(
+        self, component, git_repo, version, arch, snapshot, gradle_project=None
+    ):
+        super().__init__(component, git_repo, version, arch, snapshot)
+        self.gradle_project = gradle_project
+        self.dependencies = self.__get_dependencies()
+
+    def __get_dependencies(self):
+        cmd = " ".join(
+            [
+                f"./gradlew {self.gradle_project or ''}:dependencies",
+                f"-Dopensearch.version={self.opensearch_version}",
+                f"-Dbuild.snapshot={str(self.snapshot).lower()}",
+                '| grep -e "---"',
```

Comment:
Just trying to understand:
Looks like we are only checking opensearch and its version. Why not just look for compile time dependencies instead of all of them? 

Also for next steps, we should verify plugin dependencies on other plugins and make sure their versions also match the manifest.

Reply:
Do you know how to get just those? I can open a separate PR for improving this.

Reply:
https://github.com/opensearch-project/opensearch-build/issues/440 and https://github.com/opensearch-project/opensearch-build/issues/441

Reply:
Yeah a simple way is to do something like `./gradlew :dependencies --configuration compile`

Eg: From our fav plugin AD
```
~anomaly-detection$ ./gradlew :dependencies --configuration compile

> Configure project :
=======================================
OpenSearch Build Hamster says Hello!
  Gradle Version        : 6.6.1
  OS Info               : Linux 5.4.0-1037-aws (amd64)
  JDK Version           : 14 (JDK)
  JAVA_HOME             : /usr/lib/jvm/java-14-openjdk-amd64
  Random Testing Seed   : DFA8FDF8CA53A245
  In FIPS 140 mode      : false
=======================================

> Task :dependencies

------------------------------------------------------------
Root project - OpenSearch anomaly detector plugin
------------------------------------------------------------

compile - Dependencies for source set 'main' (deprecated, use 'implementation' instead). (n)
+--- org.opensearch:opensearch:1.0.0-rc1 (n)
+--- org.opensearch:common-utils:1.0.0.0-rc1 (n)
+--- com.google.guava:guava:29.0-jre (n)
+--- org.apache.commons:commons-math3:3.6.1 (n)
+--- com.google.code.gson:gson:2.8.6 (n)
+--- com.yahoo.datasketches:sketches-core:0.13.4 (n)
+--- com.yahoo.datasketches:memory:0.12.2 (n)
+--- commons-lang:commons-lang:2.6 (n)
+--- software.amazon.randomcutforest:randomcutforest-core:1.0 (n)
+--- software.amazon.randomcutforest:randomcutforest-serialization-json:1.0 (n)
+--- org.opensearch.client:opensearch-rest-client:1.0.0-rc1 (n)
+--- org.jacoco:org.jacoco.agent:0.8.5 (n)
\--- org.jacoco:org.jacoco.ant:0.8.5 (n)
```